pkgs/by-name/cm/cmake/setup-hook.sh

```diff
@@ -1,5 +1,14 @@
 addCMakeParams() {
-    addToSearchPath CMAKE_PREFIX_PATH $1
+    addToSearchPath CMAKE_INCLUDE_PATH "$1/include"
```

Comment:
https://cmake.org/cmake/help/latest/variable/CMAKE_INCLUDE_PATH.html#variable:CMAKE_INCLUDE_PATH says "semi-colon separated", so I think we need

```suggestion
    addToSearchPathWithCustomDelimiter CMAKE_INCLUDE_PATH ";" "$1/include"
```

I wonder if it'd make sense to add an alias, something like `addCmakeSearchPath`, because this construct will already be repeated in `cmake/setup-hooks.h` and we're also using this in `pkgs/development/cuda-modules/setup-hooks/setup-cuda-hook.sh`

Reply:
> https://cmake.org/cmake/help/latest/variable/CMAKE_INCLUDE_PATH.html#variable:CMAKE_INCLUDE_PATH says "semi-colon separated", so I think we need
...

you're looking at the wrong page. that is not the environment variable but the cmake variable. https://cmake.org/cmake/help/latest/envvar/CMAKE_INCLUDE_PATH.html#envvar:CMAKE_INCLUDE_PATH

it is not semi-colon separated but normal unix colon separated -- unless on windows.